src/balances.lua

```diff
@@ -20,15 +25,24 @@ function balances.transfer(recipient, from, qty)
 	}
 end
 
+--- Gets the balance for a specific address
+---@param target string The address to get balance for
+---@return number The balance amount (0 if address has no balance)
 function balances.getBalance(target)
-	return utils.deepCopy(Balances[target] or 0)
+	return Balances[target] or 0
 end
 
+--- Gets all balances in the system
+---@return table All address:balance pairs
 function balances.getBalances()
-	local balances = utils.deepCopy(Balances)
-	return balances or {}
+	local balances = utils.deepCopy(Balances) or {}
+	return balances
 end
 
+--- Reduces the balance of an address
+---@param target string The address to reduce balance for
+---@param qty number The amount to reduce by (must be integer)
+---@throws error If target has insufficient balance
 function balances.reduceBalance(target, qty)
 	local prevBalance = balances.getBalance(target)
```

Comment:
This is likely another unnecessary copy of ALL the balances just to update one value in place.